test/framework/src/main/java/org/elasticsearch/index/shard/IndexShardTestCase.java

```diff
@@ -265,24 +292,26 @@ protected IndexShard newShard(ShardRouting routing, IndexMetaData indexMetaData,
         final NodeEnvironment.NodePath nodePath = new NodeEnvironment.NodePath(createTempDir());
         ShardPath shardPath = new ShardPath(false, nodePath.resolve(shardId), nodePath.resolve(shardId), shardId);
         return newShard(routing, shardPath, indexMetaData, indexSearcherWrapper, engineFactory, globalCheckpointSyncer,
-            EMPTY_EVENT_LISTENER, listeners);
+            EMPTY_EVENT_LISTENER, false, listeners);
     }
 
     /**
      * creates a new initializing shard.
-     *  @param routing                shard routing to use
+     * @param routing                shard routing to use
      * @param shardPath              path to use for shard data
      * @param indexMetaData          indexMetaData for the shard, including any mapping
      * @param indexSearcherWrapper   an optional wrapper to be used during searchers
      * @param globalCheckpointSyncer callback for syncing global checkpoints
      * @param indexEventListener     index even listener
+     * @param ignoreShardFailures    true if shard failure should be ignored (otherwise an assertion will be thrown)
      * @param listeners              an optional set of listeners to add to the shard
      */
     protected IndexShard newShard(ShardRouting routing, ShardPath shardPath, IndexMetaData indexMetaData,
                                   @Nullable IndexSearcherWrapper indexSearcherWrapper,
                                   @Nullable EngineFactory engineFactory,
                                   Runnable globalCheckpointSyncer,
-                                  IndexEventListener indexEventListener, IndexingOperationListener... listeners) throws IOException {
+                                  IndexEventListener indexEventListener, boolean ignoreShardFailures,
```

Comment:
is this parameter used anywhere?

Reply:
left over of a failed attempt, will remove. good catch